app.js

```diff
@@ -23,63 +34,144 @@ var stuff = [
 
-var pictureBlockEl = document.getElementById('productPictures');
-function postPictures(){
-  var pictureEl = document.createElement('h1');
-  pictureBlockEl.appendChild(pictureEl);
+function displayPics(){
+  var leftIndex = findRandom();
+  var centerIndex = findRandom();
+  var rightIndex = findRandom();
 
-  for(var i = 0; i < randomNumArray.length; i++){
-    var addPic = document.createElement('img');
-    addPic.setAttribute('src', randomNumArray[i]);
-    pictureEl.appendChild(addPic);
+  // console.log('starting left index', leftIndex);
+  // console.log('starting center index', centerIndex);
+  // console.log('starting right index', rightIndex);
 
-    console.log(typeof(randomNumArray[i]));
+  while(currentlyShowing.includes(leftIndex)){
+    leftIndex = findRandom();
+    // console.log('new left', leftIndex);
   }
-}
+  while(centerIndex === leftIndex || currentlyShowing.includes(centerIndex)){
+    centerIndex = findRandom();
+    // console.log('new center', centerIndex);
+  }
+  while(rightIndex === leftIndex || rightIndex === centerIndex || currentlyShowing.includes(rightIndex)){
+    rightIndex = findRandom();
+    // console.log('new right', rightIndex);
+
+  }
+  // console.log(leftIndex, centerIndex, rightIndex);
+  stuff[leftIndex].views ++;
+  stuff[centerIndex].views ++;
+  stuff[rightIndex].views ++;
+  // console.log(stuff[3].views, stuff[3].name);
+
+  pictureBlockEl.removeChild(left);
+  left = document.createElement('img');
+  left.setAttribute('src', stuff[leftIndex].src);
+  pictureBlockEl.appendChild(left);
+
+  pictureBlockEl.removeChild(center);
+  center = document.createElement('img');
+  center.setAttribute('src', stuff[centerIndex].src);
+  pictureBlockEl.appendChild(center);
+
+  pictureBlockEl.removeChild(right);
+  right = document.createElement('img');
+  right.setAttribute('src', stuff[rightIndex].src);
+  pictureBlockEl.appendChild(right);
 
-function removePictures(){
-  var pictureEl = document.getElementById('prodcutPictures');
-  var removePicEl = document.getElementByTag('h1');
-  pictureEl.removeChild(removePicEl);
+  currentlyShowing = [leftIndex, centerIndex, rightIndex];
 }
-postPictures();
 
-console.log('-------------------Event Listener click---------------');
+displayPics();
+
+console.log('-----------------Event Listener click---------------');
 
 var onImageClickEl = document.getElementById('productPictures');
 
 onImageClickEl.addEventListener('click', handleClick);
 
-//event listener not working corectly yet. Needs to delete current picure and add new
 function handleClick(event){
+
   event.preventDefault();
   event.stopPropagation();
 
-  removePictures();
+  var target = event.target;
+  var targetSrc = target.getAttribute('src');
+
+  if(totalClicks < clickLimit){
+
+    findRandom();
+    displayPics();
+
+    totalClicks ++;
+    console.log('click number ' + totalClicks);
+    // console.log('currentlyShowing ' + currentlyShowing);
+    for(i = 0; i < stuff.length; i++){
```

Comment:
You should write a comment here to describe what's going on, and maybe you could consider breaking this out into a function?